src/tasks/Promotions/YouTube/Subscription.ts

```diff
@@ -0,0 +1,104 @@
+import nodeFetch, { Response } from 'node-fetch';
+import * as FormData from 'form-data';
+import { Channel, DataService } from './DataService';
+
+export class YouTubeSubscription {
+  constructor() {
+    YouTubeSubscription.run();
+  }
+  static async sendPshbRequest(channelId: string, mode: string): Promise<Response | void> {
+    try {
+      const hostname = process.env.HOSTNAME;
+
+      if (!hostname) {
+        throw {
+          title: 'PubSubHubBub Error',
+          message: 'Hostname is needed to provide valid callback function'
+        }
+      }
+      
+      const pshbUrl = 'https://pubsubhubbub.appspot.com';
+      const topicUrl = 'https://www.youtube.com/xml/feeds/videos.xml?channel_id=';
+      const callbackUrl = `https://${hostname}/youtube/notification`;
+    
+      const form = new FormData();
+    
+      form.append('hub.callback', callbackUrl);
+      form.append('hub.topic', topicUrl + channelId);
+      form.append('hub.mode', mode);
+    
+      return await nodeFetch(pshbUrl, {
+        method: 'POST',
+        body: form,
+      });
+
+    } catch(e) {
+      console.error(e);
+    }
+  }
+
+  /**
+   * Get status subscription for given channel. Status available on
+   * PubSubHubBub official webpage. Access it by GET request with
+   * query parameters of callbackUrl and topicUrl and parse this page
+   * with RegExp. Statuses: 'verified', 'unverified' and 'unsubscribed'.
+   * @param channelId 
+   */
+  static async getStatus(channelId: string): Promise<string | void> {
+    try {
+
+      const hostname = process.env.HOSTNAME;
+
+      if (!hostname) {
+        throw {
+          title: 'PubSubHubBub Error',
+          message: 'Hostname is needed to provide valid callback function'
+        }
+      }
+      
+      const pshbUrl = 'https://pubsubhubbub.appspot.com/subscription-details';
+      const topicUrl = 'https://www.youtube.com/xml/feeds/videos.xml?channel_id=';
+      const callbackUrl = `https://${hostname}/youtube/notification`;
+      const url = `${pshbUrl}?hub.callback=${callbackUrl}&hub.topic=${topicUrl}${channelId}`;
+      const response = await nodeFetch(url);
+
+      if (typeof response !== 'undefined') {
+        response.body.on('data', data => {
```

Comment:
since this returns "text/html", another way we could do this is to use `response.text()`, but this is fine.